### ISSUE TITLE
Image block: images stretched on smaller devices unless a scale behaviour is set. If the selection is Contain, large margins are added. 

### ISSUE THREAD

**Issue by filipanoscampos** [Automattic/wp-calypso#79533]
### Quick summary

Recently, the Image Block received a much-welcomed scaling setting. 
Yet, there appear to be some unexpected results.

If the image is resized and the user does not choose a scaling behavior, the image is stretched on mobile devices.
If the user chooses the option to contain the image size, there are large white margins added above and below the image on small devices.

### Steps to reproduce

1. Add an image block to a page or post, preferably win-between two Paragraph blocks
2. Resize the image
3. Save your changes and visit the post from a small device (less than 780px). Notice the stretched image.
![Screenshot 2023-07-18 at 12 14 26](https://github.com/Automattic/wp-calypso/assets/2540430/91f7488c-1f55-4f0f-a658-c93ea5fe98ad)

![Screenshot 2023-07-18 at 12 14 16](https://github.com/Automattic/wp-calypso/assets/2540430/2ed822fe-fdfa-4cdb-b0a3-54b5d0e7a754)

4. Now, set the image to Contain and save your changes. Notice the large margins when the post is visited from a small device.
![Screenshot 2023-07-18 at 12 14 38](https://github.com/Automattic/wp-calypso/assets/2540430/ab2e0535-9be7-463c-889b-cd083ba74bd3)

![Screenshot 2023-07-18 at 12 08 43](https://github.com/Automattic/wp-calypso/assets/2540430/883c7f0f-a4dc-48c1-97c5-5b9755d19ce0)

### What you expected to happen

The image to adjust to the smaller width, without stretching or extra margins.

### What actually happened

The image is stretched or, if we set it to contain, there is a large white margin above and below the image.

### Impact

Most (> 50%)

### Available workarounds?

Yes, difficult to implement

### Platform (Simple and/or Atomic)

Simple and Atomic

### Logs or notes

Available workarounds:
- Using the Classic Block to add an image
- Some CSS that @Robertght kindly shared:

```
@media screen and (max-width: 768px){
 .wp-block-image img {
    height: auto !important;
 }
}
```

First reportedon this interaction 6549118-zd-woothemes
Related conversation, here: p1689669069885049-slack-C03TY6J1A

_Update - see easier [workaround in comments](https://github.com/Automattic/wp-calypso/issues/79533#issuecomment-1640625154)_

**Comment by cuemarie** [comment]
📌 SCRUBBING : RESULT - *Replicated / Could Not Replicate / Uncertain*
- Tested on Simple – **Replicated**
- Tested on Atomic – **Replicated**
- Replicable outside of Dotcom – **Yes**

📌 FINDINGS/SCREENSHOTS/VIDEO
- Replicated as described across WordPress.com and WordPress.org test sites.

#### Screenshots/Recordings

https://github.com/Automattic/wp-calypso/assets/27249804/06d136a9-4556-46a9-b1ee-8c68e29dd034

Replicated in Gutenberg 16.2.1, 16.2.0, 16.1.2, and with Gutenberg deactivated.

📌 ACTIONS
- Assigned to Other
- Added to ‘HE Cross-repo Watchlist’ project board

**Comment by cuemarie** [comment]
Reported here: https://github.com/WordPress/gutenberg/issues/52739

**Comment by cuemarie** [comment]
### Workaround

In the editor, clear out the **Height** selection and it will default to `Auto` and remove the scaling feature.  This will correct the issue:

**Editor**
![Markup on 2023-07-18 at 10:04:49](https://github.com/Automattic/wp-calypso/assets/27249804/e6cea7a7-581b-4fe7-bbca-9d94c17bcc7e)

**Live Site**
![Markup on 2023-07-18 at 10:05:11](https://github.com/Automattic/wp-calypso/assets/27249804/a6365b74-22d1-41b7-8753-7ad2e5771683)

**Comment by Robertght** [comment]
It's likely the change is caused by this: p7DVsv-hV9-p2 

@sgomes  can you confirm this is the same case? In this case, until stage 3, the users cannot wait for images to be stretched on mobile.

**Comment by sgomes** [comment]
Hey @Robertght 👋

I can't seem to reproduce the issue, I'm afraid 😕 Does this require a particular theme?

Without any further information, I doubt that the issue is related to the rollout in p7DVsv-hV9-p2, since that is not affecting any user blogs at the moment, but I'd like to make sure.

**Comment by sgomes** [comment]
Thank you to @filipanoscampos for a test case 👍

This seems to be unrelated to the rollout in p7DVsv-hV9-p2, as I expected. It looks like an issue with how this feature is implemented in Gutenberg.

The element has the following styles:
```css
{
    width: 545px;
    height: 472px;
}
```

But Gutenberg also adds the following:
```css
.wp-block-image img {
    box-sizing: border-box;
    height: auto;
    max-width: 100%;
    vertical-align: bottom;
}
```

This means that when `max-width` kicks in, the height is no longer proportional, because it’s using the `472px` fixed value, which has precedence over the `auto` because it’s in an element style attribute.

**Comment by kelasante** [comment]
Another report in 6547432-zen

**Comment by Robertght** [comment]
Another in #6589094-zen

**Comment by jp-imagines** [comment]
Another report in 6604157-zd-woothemes, though I wasn't able to replicate in Firefox on Android.

This user was reportedly able to resolve things by changing the "Scale" setting from Cover to Contain and choosing a different aspect ratio.

**Comment by cuemarie** [comment]
Looks like a fix for this will be incoming with Gutenberg 16.5 in https://github.com/WordPress/gutenberg/pull/53274

**Comment by wpshellbelle** [comment]
6641104-zd

**Comment by ClassicRKR27** [comment]
6642481-zd-a8c

**Comment by wpshellbelle** [comment]
6640399-zd

**Comment by Robertght** [comment]
6689829-zen

**Comment by jp-imagines** [comment]
Reported in 6799728-zd-a8c

**Comment by Aurorum** [comment]
7139337-zen

**Comment by Aurorum** [comment]
Further reports here - do we know when the fix is to be rolled out or if it'll be retrospective to affected blocks?

- 7138606-zen
- 7139337-zen

**Comment by cuemarie** [comment]
@Aurorum Thanks for that ask!  

I just retested on a new Free Site in WordPress.com, and the issue is no longer present, so the fix has landed for new sites.


https://github.com/Automattic/wp-calypso/assets/27249804/1dc4f1fc-c090-4d2b-bd29-b44ba767db90

---

As for whether it's expected to retroactively apply to blocks, I'm not sure. I'll do some digging today to see what I can find out.

**Comment by cuemarie** [comment]
@Aurorum Follow up - it looks like the fix applies to new image blocks, but existing image blocks that were created while this issue was in effect will need to be corrected. That's because when the image was created and resized, a height was applied to the image in the editor - upgrading Gutenberg past 16.5 (where the fix landed) does not make changes to those blocks.

You can see an example of this test on a self-hosted site below, along with the manual correction needed:


https://github.com/Automattic/wp-calypso/assets/27249804/84cc2128-9b71-4666-b77d-694e0df97ec3

### TL;DR

The bug is fixed, but users will need to edit their image blocks and remove the height setting, which will clear that height and set it back to Auto.  Once this is done, the CSS workaround is no longer needed.

📌 ACTIONS
- Closed as resolved (by: https://github.com/WordPress/gutenberg/pull/53274 )

**Comment by Aurorum** [comment]
Thanks @cuemarie - much appreciated.
